File: Leet Code/121_best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size()==0) return 0;
        int mx = 0;
        int mi = prices[0];
        for(const auto i:prices){
            mx = max(mx,i-mi);
            mi = min(mi,i);
        }
        return mx;
    }
};

File: Leet Code/121_best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mx_diff = 0
        if len(prices) == 0:
            return 0
        mi_element = prices[0]
        
        for i in range(1,len(prices)):
            if prices[i] - mi_element > mx_diff:
                mx_diff = prices[i] - mi_element 
            
            if prices[i] < mi_element:
                mi_element = prices[i]
            
        return mx_diff
        